server/src/api/task.ts

```diff
@@ -12,24 +12,61 @@ router.get('/', authenticateToken, async (req: AuthorizedRequest, res) => {
   res.json(rows);
 });
 
+const MIN_IMPORTANCE = 1;
 const MAX_IMPORTANCE = 5;
 const validateImportance = (importance: number) => {
-  return importance <= MAX_IMPORTANCE;
+  return typeof importance === 'number' && importance >= MIN_IMPORTANCE && importance <= MAX_IMPORTANCE;
 };
 const validateLatitude = (lat: number) => {
-  return lat >= -90 && lat <= 90;
+  return typeof lat === 'number' && lat >= -90 && lat <= 90;
 };
 const validateLongitude = (lng: number) => {
-  return lng >= -180 && lng <= 180;
+  return typeof lng === 'number' && lng >= -180 && lng <= 180;
+};
+
+const DEFAULT_TAG_INDEX = 1;
+
+const validateTitle = (title: string) => {
+  return typeof title === 'string';
+};
+const validateDate = (date: string) => {
+  const regex = /^\d{4}-(0[1-9]|1[012])-(0[1-9]|[12][0-9]|3[01])$/;
+  return regex.test(date);
+};
+const validateIsPublic = (isPublic: boolean) => {
+  return typeof isPublic === 'boolean';
+};
+const validateTime = (time: string) => {
+  const regex = /^([01][0-9]|2[0-3]):([0-5][0-9])$/;
+  return regex.test(time);
+};
+const validateTagIdx = (tagIdx: number) => {
+  return typeof tagIdx === 'number';
+};
+const validateDone = (done: boolean) => {
+  return typeof done === 'boolean';
+};
+// TODO: implement
+const validateLabels = (labels: any[]) => {
+  return false;
 };
 
 router.post('/', authenticateToken, async (req: AuthorizedRequest, res) => {
   const { userIdx } = req.user;
-  const { title, importance, startedAt, endedAt, lat, lng, isPublic, tagIdx, content, done, date, labels } = req.body;
+  let { title, date, importance, startedAt, endedAt, lat, lng, isPublic, tagIdx, content, done, labels } = req.body;
 
-  if (!validateImportance(importance)) res.sendStatus(400);
-  if (!validateLatitude(lat)) res.sendStatus(400);
-  if (!validateLongitude(lng)) res.sendStatus(400);
+  if (!validateTitle(title)) return res.status(400).send({ msg: '제목을 입력해주세요.' });
+  if (!validateDate(date)) return res.status(400).send({ msg: '날짜를 입력해주세요.' });
+  if (!validateImportance(importance)) importance = (MIN_IMPORTANCE + MAX_IMPORTANCE) / 2;
+  if (!validateTime(startedAt)) startedAt = null;
+  if (!validateTime(endedAt)) endedAt = null;
+  if (!validateLatitude(lat)) lat = null;
+  if (!validateLongitude(lng)) lng = null;
+  if (!validateIsPublic(isPublic)) isPublic = false;
+  if (!validateTagIdx(tagIdx)) tagIdx = DEFAULT_TAG_INDEX;
+  if (!validateLongitude(content)) content = null;
+  if (!validateDone(done)) done = false;
+  if (!validateLabels(labels)) labels = [];
```

Comment:
수고하셨습니다 꼼꼼하게 다 검증해주는거 좋네요!
굳이 피드백 하자면 뭔가 반복되는 부분을 줄이면 좋을것같다는 생각이 드네요.
default값을 상수화해두고 key value 값을 이용해보면 어떨까 싶었습니다.
```suggestion
// constants/index.js
const default = {
  startedAt: null,
  endedat: null,
  lat: null,
  ..
  isPublic: false
}

//함수 선언부
function validate(key, value) {
  if(key === 'lng'){
    return typeof value === "number" && value >= -90 && value <= 90;
  } else if (key==='title'){
    return typeof value === "string";
    }
    ..
}

//router.post 내부
object.entries(req.body).forEach((key, value)=>{
   if(!validate(key, value)) req.body[key] = default[key]
})
}

const { title, importance, startedAt, endedAt, lat, lng, isPublic, tagIdx, content, done, date, labels } = req.body;
```

Reply:
피드백 반영했습니다! 감사합니다.
https://github.com/boostcampwm-2022/web19-Boostart/pull/89/commits/2e6ddfbc627c132069d1d68ba72498576552c1b5